### PR TITLE
Setup YAML pipeline for docs build

### PR DESCRIPTION
**Pull request by skirby1996**
Replaces the classic Docs CI build with a YAML based pipeline for all releases later than 2.x.

### CODE REVIEW

File: common/config/azure-pipelines/jobs/docs-ci.yaml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    pool:
+      name: iModelTechCI
+      demands: Agent.OS -equals Windows_NT
+
+    steps:
+      - checkout: self
+        clean: true
+
+      - task: NodeTool@0
+        displayName: Use Node 12.17.0
```
Comment (calebmshafer) [inline common/config/azure-pipelines/jobs/docs-ci.yaml:35]: just a note, we should update this.

**Review by calebmshafer — APPROVED** [review]
This all looks good to me but I'd like @williamkbentley to take a look before merging.

File: common/config/azure-pipelines/templates/docs-build.yaml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  - name: downloadCurrentBuildArtifacts
+    type: boolean
+    default: false
+
+steps:
+  - script: node common/scripts/install-run-rush.js install
+    displayName: rush install
+    workingDirectory: ${{ parameters.workingDir }}
+
+  - script: node common/scripts/install-run-rush.js build
```
Comment (williamkbentley) [inline common/config/azure-pipelines/templates/docs-build.yaml:23]: Should we use `build:ci`?

File: common/config/azure-pipelines/templates/gather-docs.yaml [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+      artifactName: Bis Docs
+    condition: and(succeeded(), eq('${{ parameters.downloadCurrentBuildArtifacts }}', true))
+
+  - task: CopyFiles@2
+    displayName: Copy Bis Docs to staging
+    inputs:
+      SourceFolder: "$(System.ArtifactsDirectory)/Bis Docs"
+      TargetFolder: ${{ parameters.stagingDir }}/bis/domains/
+
+  # The .updated.json is cached in order to let BeMetalsmith know what needs to be re-built.
```
Comment (williamkbentley) [inline common/config/azure-pipelines/templates/gather-docs.yaml:67]: Not accurate. updated.json is used for the "Last Updated" date on the footer of all docs

File: common/config/azure-pipelines/jobs/docs-ci.yaml [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+          checkLatest: true
+
+      - script: |
+          git config --local user.email imodeljs-admin@users.noreply.github.com
+          git config --local user.name imodeljs-admin
+        displayName: Setup git config
+
+      - template: ../templates/docs-build.yaml
+
+      - publish: $(Agent.BuildDirectory)/tempDocsBuild/
```
Comment (williamkbentley) [inline common/config/azure-pipelines/jobs/docs-ci.yaml:47]: Is this enough to publish? I think the release pipeline expects a nuget package

**Review by williamkbentley — COMMENTED** [review]
Few comments and suggestions

File: common/config/azure-pipelines/jobs/docs-ci.yaml (hunk 4 above)
Comment (skirby1996) [reply common/config/azure-pipelines/jobs/docs-ci.yaml:47]: I've moved cred scanning and nuget packaging from the docs CI build into the the release pipeline. I've added a duplicate series of deployments steps postfixed with "YAML test" for testing with this new artifact, you should be able to see logs in [this release](https://dev.azure.com/bentleycs/beconnect/_releaseProgress?_a=release-pipeline-progress&releaseId=166347).

**Review by williamkbentley — APPROVED** [review]
LG

**Pull request merged**